### PR TITLE
[Enhancement] Adding Markdown files for Regex, Contains and Exact.

### PR DESCRIPTION
**Pull request by dawnvt**
This is marked as a Draft PR untill I finish the rest 
* regex/README.md explains how to use Regex in a more efficient way in Boost Regex and LookupID. 
* contains/README.md explains how the Contains method works, and how to use it
* exact/README.md explains how the Exact method works, and how to use it.

### CODE REVIEW

File: how-tos/regex/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+`[:blank:]` - any blank character (space and tab) <br>
+
+
+Amount modifiers: <br>
+`*` - 0 or more characters <br>
+`+` - 1 or more characters  <br>
+`{1,3}` - 1 to 3 characters <br>
+
+OR:
+`(a|b)` - matches both "a" and "b"
```
Comment (Omurky) [inline how-tos/regex/README.md:27]: matches "a" or "b"

File: how-tos/regex/README.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+Using [this tool](https://regex101.com/) helps with learning Regex a lot faster. But be aware that you should still be comfortable with Regex before its used in an actual map.
+
+With that out of the way, here is all the important stuff:
+
+Characters: <br>
+`.` - any character <br>
+`[a-z]` - single character from a to z <br>
+`[a-z0-9A-Z]` - single character from a to z, 0 to 9 and A to Z <br>
+`[:blank:]` - any blank character (space and tab) <br>
+
```
Comment (Omurky) [inline how-tos/regex/README.md:19]: `abcd` - plainly matches the string "abcd"

File: how-tos/regex/README.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+Amount modifiers: <br>
+`*` - 0 or more characters <br>
+`+` - 1 or more characters  <br>
+`{1,3}` - 1 to 3 characters <br>
+
+OR:
+`(a|b)` - matches both "a" and "b"
+
+Examples: <br>
+`[0-9]+,[0-9]+` - one or more digits followed by a comma followed by one or more digits (412,66 would match, 5a,2 wouldn't) <br>
```
Comment (Omurky) [inline how-tos/regex/README.md:30]: ("412,66" would match, "5a,2" would not)

File: how-tos/regex/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+`[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}` - Poor-man's ipv4 regex (matches 4 groups of [0-9]{1,3} separated by dots). <br>
+`ba+d` - matches "bad", "baad", "baaad" etc. <br>
+`b(a|e)d` - matches "bad" and "bed"
+
+These are just **examples** of special characters, amounts or clauses. You **will** need the [documentation](https://www.boost.org/doc/libs/1_31_0/libs/regex/doc/syntax.html) to make the best use of Boost Regex.
+
+This following example will return all 3 instances of `Pillar` appearing, first in PillarPair, then PillarL, then PillarR.
```
Comment (Omurky) [inline how-tos/regex/README.md:37]: Order makes no difference, which the wording "first [..], then [..]" implies. I'd reword it slightly:

[..] appearing: PillarPair, PillarL and PillarR.

**Review by Omurky — CHANGES_REQUESTED** [review]
Made some comments no the regex readme.

File: how-tos/regex/README.md (hunk 4 above)
Comment (dawnvt) [reply how-tos/regex/README.md]: 36c1df5

File: how-tos/regex/README.md (hunk 3 above)
Comment (dawnvt) [reply how-tos/regex/README.md]: 36c1df5

File: how-tos/regex/README.md (hunk 2 above)
Comment (dawnvt) [reply how-tos/regex/README.md:20]: 36c1df5

File: how-tos/regex/README.md (hunk 1 above)
Comment (dawnvt) [reply how-tos/regex/README.md]: 36c1df5

**Review by dawnvt — COMMENTED** [review]
36c1df5

**Review by Omurky — APPROVED** [review]
:+1:

**Review by StormPacer — COMMENTED** [review]
Looks good
